### PR TITLE
Action Cable: JavaScript test updates

### PR DESCRIPTION
**Pull request by javan**
* Adds more JavaScript tests (cc: @maclover7)
* Ensures Sauce Labs tunnel closes properly and fixes CI errors like https://travis-ci.org/rails/rails/jobs/165766716#L506 (cc: @matthewd)

### CODE REVIEW

File: actioncable/test/javascript/src/unit/subscription_test.coffee [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    server.on "message", (message) ->
+      data = JSON.parse(message)
+      assert.equal data.identifier, current.subscription.identifier
+      assert.equal data.command, "message"
+      assert.deepEqual data.data, JSON.stringify(action: { publish: "hi" })
+      done()
+
+    broadcast(message_type: "confirmation")
+
+
```
Comment (maclover7) [inline actioncable/test/javascript/src/unit/subscription_test.coffee]: too many empty lines here? 😬

File: actioncable/test/javascript/src/unit/subscription_test.coffee [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+broadcast = (data = {}, callback) ->
```
Comment (maclover7) [inline actioncable/test/javascript/src/unit/subscription_test.coffee]: Might be a bit early, but should we extract the `broadcast` function out to where the rest of the "helper" functions are?

**Review by maclover7 — CHANGES_REQUESTED** [review]
Overall big 👍 👍 to adding more tests for Action Cable's JS! If possible, can you make each of the test's names more descriptive? While I was reading through, it was a little tough to determine what each of the tests was doing 😬

**Comment by javan** [comment]
@maclover7 done and done

**Comment by kaspth** [comment]
Nice!

**Pull request merged**